pkg/webhook/mutation/pod/otlp/mutator.go

```diff
@@ -0,0 +1,234 @@
+)
+
+const (
+	envVarOtlpTraceEndpoint    = "OTEL_EXPORTER_OTLP_TRACES_ENDPOINT"
+	envVarOtlpLogsEndpoint     = "OTEL_EXPORTER_OTLP_LOGS_ENDPOINT"
+	envVarOtlpMetricsEndpoint  = "OTEL_EXPORTER_OTLP_METRICS_ENDPOINT"
+	envVarOtlpExporterEndpoint = "OTLP_EXPORTER_OTLP_ENDPOINT"
+
+	envVarOtlpTraceProtocol    = "OTEL_EXPORTER_OTLP_TRACES_PROTOCOL"
+	envVarOtlpLogsProtocol     = "OTEL_EXPORTER_OTLP_LOGS_PROTOCOL"
+	envVarOtlpMetricsProtocol  = "OTEL_EXPORTER_OTLP_METRICS_PROTOCOL"
+	envVarOtlpExporterProtocol = "OTLP_EXPORTER_OTLP_PROTOCOL"
+
+	envVarOtlpTraceHeaders    = "OTEL_EXPORTER_OTLP_TRACES_HEADERS"
+	envVarOtlpLogsHeaders     = "OTEL_EXPORTER_OTLP_LOGS_HEADERS"
+	envVarOtlpMetricsHeaders  = "OTEL_EXPORTER_OTLP_METRICS_HEADERS"
+	envVarOtlpExporterHeaders = "OTLP_EXPORTER_OTLP_HEADERS"
+
+	envVarOtlpTraceCertificate    = "OTEL_EXPORTER_OTLP_TRACES_CERTIFICATE"
+	envVarOtlpLogsCertificate     = "OTEL_EXPORTER_OTLP_LOGS_CERTIFICATE"
+	envVarOtlpMetricsCertificate  = "OTEL_EXPORTER_OTLP_METRICS_CERTIFICATE"
+	envVarOtlpExporterCertificate = "OTLP_EXPORTER_OTLP_CERTIFICATE"
+
+	// TODO make sure naming does not collide with anything
+	activeGateTrustedCertVolumeName    = "activegate-server-certs"
+	activeGateTrustedCertSecretKeyPath = "activegate.pem"
+	activeGateTrustedCertMountPath     = "activegate-server-certs"
+	envActiveGateTrustedCert           = activeGateTrustedCertMountPath + "/" + activeGateTrustedCertSecretKeyPath
+
+	envVarDtApiToken = "DT_API_TOKEN"
+
+	otlpAPIPath = "/v2/otlp"
+)
+
+type OTLPMutator struct{}
+
+func NewMutator() dtwebhook.Mutator {
+	return &OTLPMutator{}
+}
+
+func (OTLPMutator) IsEnabled(request *dtwebhook.BaseRequest) bool {
+	//TODO implement when structure for OTLP enrichment config in Dynakube has been defined.
+	return true
+}
+
+func (OTLPMutator) IsInjected(request *dtwebhook.BaseRequest) bool {
+	// TODO the check for already present env vars is also done in the Mutate() implementation, so I'm not sure we need this check here as well.
+	// Maybe a check for an annotation indicating that the OTLP vars have already been injected might be better here.
+	otlpEndpointEnvVars := []string{envVarOtlpExporterEndpoint, envVarOtlpTraceEndpoint, envVarOtlpMetricsEndpoint, envVarOtlpLogsEndpoint}
+	// if any of the OTEL_EXPORTER_* variables are already set in all containers, return false, as we do not want to overwrite them
+	// TODO check if also initContainers should be considered
+	gotOtlpExporterEnvVars := 0
+
+	for _, c := range request.Pod.Spec.Containers {
+		found := false
+		for _, e := range c.Env {
+			// check if any of the otlp exporter env vars is already present in the container
+			if slices.Contains(otlpEndpointEnvVars, e.Name) {
+				found = true
+				break
+			}
+		}
+		if found {
+			gotOtlpExporterEnvVars++
+		}
+	}
+	if gotOtlpExporterEnvVars == len(request.Pod.Spec.Containers) {
+		return true
+	}
+	return false
+}
+
+func (OTLPMutator) Mutate(request *dtwebhook.MutationRequest) error {
```

Comment:
here we use `OTLP`, and in different places we use Otlp, so please stick with one name convention, we prefer and enforce `OTLP` since it's abbrv